Pseudo/css/pseudoElements.css

```diff
@@ -0,0 +1,101 @@
+/*FirstLetter*/
+
+.firstLetter p::first-letter {
+	color: red;
+	font-size: 23px;
+}
+
+/*firstLine*/
+
+.firstLine p::first-line {
+	color: green;
+}
+
+/*selection*/
+
+.selection p::selection {
+	background-color: yellow;
+}
+
+/*beforeafterStyling*/
+
+.beforeafterStyling {
+	text-align: center;
+	overflow-x: hidden;
+}
+.beforeafterStyling h3{
+	text-align: left;
+}
+.beforeafterStyling h4 {
+	display: inline-block;
+	position: relative;
+}
+.beforeafterStyling h4::before,.beforeafterStyling h4::after {
+	position: absolute;
+	content: '';
+	width: 900px;
+	height: 2px;
+	background-color: red;
+	top: 10px;
+}
+.beforeafterStyling h4::before {
+	right: 100%;
+	margin-right: 15px;
+}
+.beforeafterStyling h4::after {
+	left: 100%;
+	margin-left: 15px;
+}
+
+/*hoverEffect*/
+
+.hoverEffect a {
+	border: 1px solid #000;
+	padding:10px 15px;
+	display: inline-block;
+	position: relative;
+}
+.hoverEffect a::before,.hoverEffect a::after {
+	width: 15%;
+	height: 40%;
+	content: '';
+ 	position: absolute;
+ 	transition: all 0.4s ease-in-out;
+}
+.hoverEffect a::before {
+	border-left: 1px solid #000000;
+	border-top: 1px solid #000000;
+	left:0;
+	top: 0;
+}
+.hoverEffect a::after {
+ 	border-right: 1px solid #000000;
+ 	border-bottom: 1px solid #000000;
+ 	right: 0;
+ 	bottom: 0;
+}
+.hoverEffect a:hover:before,.hoverEffect a:hover::after {
+	width: 100%;
+	height: 100%;
+}
+
+/*borderOnhover*/
+
+.hoverBorder a {
+	padding: 10px 15px;
+	display: inline-block;
+	position: relative;
+}
+.hoverBorder a::before {
+	position: absolute;
+	content: '';
+	bottom: 0;
+	left: 0;
+	right:100%;
```

Comment:
Why left: 0 and right: 100%? Use only one of the two

Reply:
right is transion property do it and check

Reply:
Okay.

Reply:
I tried i am not getting it